packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -205,7 +205,8 @@ export class GraphNG extends Component<GraphNGProps, GraphNGState> {
           next: () => {
             const u = this.plotInstance?.current;
 
-            if (u) {
+            // @ts-ignore
+            if (u && !u.cursor._lock) {
```

Comment:
we are using uPlot's cursor locking for fixating context menus, this ensures that we don't hide a locked cursor when hovering the legend emits a DataHoverClearEvent. need the `@ts-ignore` cause `._lock` is an internal property for the current lock state.